### DISCUSSION TITLE
Multiple RaspberryPie & replication & SD card corruption

### DISCUSSION THREAD

**Discussion by npehrsson** [ravendb/ravendb#13441] [Q&A]
Hi I have two questions.

We're currently having an IoT solution where we  have 1-10 Raspberry Pie's in a network.

One of the Pie's is selected as Coordinator and communicates with the other Raspberry Pie's.
The coordinator is the one that communicates with our server in the cloud. If the coordinator goes down another Pie will be selected as Coordinator and restore the operations in the same state as the other one were.

Todo thai were synchronizing the data between the instances. We always sync the full state as it so far is in the Suze of a couple of kb.
We then store the files as files in protobuf.

As the product evolves we can now see that size of data will increase and some need for other database features is needed.

I know by been following RavenDb since it were born and have read lots of times the performamce tests that it works well on raspberry pi.

* Would RavenDb embedded work good in this scenario (my thoughts is the simplicity to upgrade because we have OTA of our application already)?

* Does RavenDb replication work good in our scenario.
Our scenario is that a Pie can go down at any second and when that happens another Pie will win an election and take over, in this scenario it will recover the state from the database/files. The questions is can we have control over which ravendb that will be master?

Worth to note the state needed for recovery is very small, most.of the data we want to store are logs.

* If the RavenDb file is damaged due to SD related issues, can it easily be recovered from other nodes. Or if only one node be somewhat repaired?

**Comment by ayende** [comment]
Yes, RavenDB can be used in this scenario.
The mode you'll likely run is a cluster of nodes, where all the Pis run the same database (fully mesh replication).
You give the cluster the task of replicating the data to the cloud and from that point on, RavenDB will take over and handle all of that, sending the data, etc.

An SD card issue can be resolved by:

* Shutting down RavenDB
* Deleting the data directory
* Restarting RavenDB

It will sync itself from the network and join up as usual.
